src/front-end/src/hooks/useDebounce.ts

```diff
@@ -0,0 +1,12 @@
+import { useState, useEffect } from "react";
+
+export function useDebounce(input: string, ms: number) {
+  const [debounced, setDebounced] = useState("");
+
+  useEffect(() => {
+    const timeout = setTimeout(() => setDebounced(input), ms);
+    return () => clearTimeout(timeout);
+  }, [input, ms]);
+
+  return debounced;
+}
```

Comment:
In general can you go through and add newlines to these files where one is missing at the end?